.github/workflows/_mac-build.yml

```diff
@@ -119,10 +125,20 @@ jobs:
           max_attempts: 3
           retry_wait_seconds: 90
           command: |
-            sudo curl --retry 3 --retry-all-errors https://s3.amazonaws.com/ossci-macos/sccache_v2.15 --output /usr/local/bin/sccache
+            if [[ "${SCCACHE_USE_GHA}" == "true" ]]; then
+              sudo curl --retry 3 --retry-all-errors -OL https://github.com/mozilla/sccache/releases/download/v0.4.0-pre.7/sccache-v0.4.0-pre.7-x86_64-apple-darwin.tar.gz
```

Comment:
Could we cache this one on S3 `ossci-macos` bucket?  I guess it's ok because it's open source.